lib/doorkeeper/config.rb

```diff
@@ -7,54 +7,42 @@ def initialize
 
   def self.configure(&block)
     @config = Config::Builder.new(&block).build
-    enable_orm
-    check_for_missing_columns
+    setup_orm_adapter
+    setup_orm_models
     setup_application_owner if @config.enable_application_owner?
+    check_requirements
   end
 
   def self.configuration
     @config || (fail MissingConfiguration.new)
   end
 
-  def self.check_for_missing_columns
-    if Doorkeeper.configuration.orm == :active_record &&
-        ActiveRecord::Base.connected? &&
-        ActiveRecord::Base.connection.table_exists?(
-          Doorkeeper::Application.table_name
-        ) &&
-        !Doorkeeper::Application.new.attributes.include?("scopes")
-
-      puts <<-MSG.squish
-[doorkeeper] Missing column: `oauth_applications.scopes`.
-If you are using ActiveRecord run `rails generate doorkeeper:application_scopes
-&& rake db:migrate` to add it.
-      MSG
-    end
+  def self.check_requirements
+    @orm_adapter.check_requirements(@config)
   end
 
-  def self.enable_orm
-    class_name = "doorkeeper/orm/#{configuration.orm}".classify
-    class_name.constantize.initialize_models!
+  def self.setup_orm_adapter
+    @orm_adapter = "doorkeeper/orm/#{configuration.orm}".classify.constantize
   rescue NameError => e
     if e.instance_of?(NameError)
-      fail e, "ORM adapter not found (#{configuration.orm})", <<-error_msg
+      fail e, "ORM adapter not found (#{configuration.orm})", <<-ERROR_MSG
 [doorkeeper] ORM adapter not found (#{configuration.orm}), or there was an error
 trying to load it.
 
 You probably need to add the related gem for this adapter to work with
 doorkeeper.
-
-If you are working on the adapter itself, double check that the constant exists,
-and that your `initialize_models!` method doesn't raise any errors.\n
```

Comment:
`initialize_models!` not call in this method, in `setup_orm_models` it will show the full error stack when `initialize_models!` raise error, so the tip is no use now.